sycl/include/sycl/ext/oneapi/experimental/chunk.hpp

```diff
@@ -51,7 +51,11 @@ template <size_t ChunkSize, typename ParentGroup> class chunk {
 
   id_type get_group_id() const {
 #ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_SubgroupLocalInvocationId() / ChunkSize;
+    if constexpr (is_chunk_v<ParentGroup>)
+      return __spirv_SubgroupLocalInvocationId() % ParentGroup::chunk_size /
+             ChunkSize;
```

Comment:
I don't want this suggestion to block anything, but I'm curious...

Do you think we should do it this way, or should we consider swapping it out for something more generic like `parent.get_local_linear_id() % ChunkSize`? The only potential downside I see is that there might be some cases where it becomes harder for the compiler to propagate some compile-time information, but I'm not sure how many such cases would actually exist in practice.

We could do something similar for the other branches.

Reply:
I think the problem with that approach is that we need to carry a copy of the parent around with us. In the current implementation it is very thin member-wise and can do everything using free functions and the compile-time information from the parent.